File: blurkit/src/main/java/io/alterac/blurkit/RoundedImageView.java
package io.alterac.blurkit;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;
import android.graphics.RectF;
import android.graphics.Xfermode;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.widget.ImageView;

public class RoundedImageView extends ImageView {

    private float mCornerRadius = 0;
    public static final int DEFAULT_COLOR = 0xff000000;
    public static final int DEFAULT_RGB = 0;

    private RectF rectF;
    private PorterDuffXfermode porterDuffXfermode;

    public RoundedImageView(Context context) {
        super(context, null);
        rectF = new RectF();
        porterDuffXfermode = new PorterDuffXfermode(PorterDuff.Mode.SRC_IN);
    }

    public RoundedImageView(Context context, AttributeSet attributes) {
        super(context, attributes);
        rectF = new RectF();
        porterDuffXfermode = new PorterDuffXfermode(PorterDuff.Mode.SRC_IN);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        Drawable myDrawable = getDrawable();
        if (myDrawable!=null && myDrawable instanceof BitmapDrawable && mCornerRadius > 0) {
            rectF.set(myDrawable.getBounds());
            int prevCount = canvas.saveLayer(rectF, null, Canvas.ALL_SAVE_FLAG);
            getImageMatrix().mapRect(rectF);

            Paint paint = ((BitmapDrawable) myDrawable).getPaint();
            paint.setAntiAlias(true);
            paint.setColor(DEFAULT_COLOR);
            Xfermode prevMode = paint.getXfermode();

            canvas.drawARGB(DEFAULT_RGB, DEFAULT_RGB, DEFAULT_RGB, DEFAULT_RGB);
            canvas.drawRoundRect(rectF, mCornerRadius, mCornerRadius, paint);

            paint.setXfermode(porterDuffXfermode);
            super.onDraw(canvas);

            paint.setXfermode(prevMode);
            canvas.restoreToCount(prevCount);
        } else {
            super.onDraw(canvas);
        }
    }

    public void setCornerRadius(float cornerRadius) {
        this.mCornerRadius = cornerRadius;
    }

    public float getCornerRadius() {
        return this.mCornerRadius;
    }
}

File: blurkit/src/main/java/io/alterac/blurkit/RoundedImageView.kt
package io.alterac.blurkit

import android.content.Context
import android.graphics.Canvas
import android.graphics.PorterDuff
import android.graphics.PorterDuffXfermode
import android.graphics.RectF
import android.graphics.drawable.BitmapDrawable
import android.util.AttributeSet
import android.widget.ImageView

class RoundedImageView : ImageView {

    private var mCornerRadius = 0f

    private var rectF: RectF? = null
    private var porterDuffXfermode: PorterDuffXfermode? = null

    companion object {
        const val DEFAULT_COLOR = -0x1000000
    }

    constructor(context: Context) : super(context, null)
    constructor(context: Context, attributes: AttributeSet) : super(context, attributes)

    init {
        rectF = RectF()
        porterDuffXfermode = PorterDuffXfermode(PorterDuff.Mode.SRC_IN)
    }

    /**
     * Draw the RoundedImageView.
     */
    override fun onDraw(canvas: Canvas) {
        val myDrawable = drawable
        if (myDrawable != null && myDrawable is BitmapDrawable && mCornerRadius > 0) {
            val paint = myDrawable.paint

            rectF!!.set(myDrawable.bounds)

            val prevCount = canvas.saveLayer(rectF, null)

            paint.isAntiAlias = true
            paint.color = DEFAULT_COLOR

            canvas.drawARGB(0, 0, 0, 0)
            canvas.drawRoundRect(rectF!!, mCornerRadius, mCornerRadius, paint)

            val prevMode = paint.xfermode
            paint.xfermode = porterDuffXfermode
            super.onDraw(canvas)

            paint.xfermode = prevMode
            canvas.restoreToCount(prevCount)
        } else {
            super.onDraw(canvas)
        }
    }

    /**
     * Set corner radius for RoundedImageView.
     */
    fun setCornerRadius(cornerRadius: Float) {
        this.mCornerRadius = cornerRadius
    }

    /**
     * Get corner radius value.
     */
    fun getCornerRadius(): Float {
        return this.mCornerRadius
    }
}